test/log-generators/install-cronjobs.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Create a CronJob that run every minute, creating an Apache log file between 128 and 512 lines long.
+# The pods for the last 1000 successful runs and 50 failed runs will be kept on the cluster, any logs
+# over those thresholds are automatically cleaned up by kubernetes.
+#
+# The job sleeps at random intervals for 1-3 seconds to get different timestamps in the log.
+
+namespace="generate-logs-cronjobs"
+
+kubectl create ns ${namespace}
+
+kubectl -n ${namespace} apply -f - << EOF
```

Comment:
I would prefer to have the YAML definition of the CronJob in a separate file, specially for resources that are larger than 10 lines (for having a threshold defined).

The main advantage I see here is to take advantage of the syntax hightlighting and, in case of YAML, easier to check potential indentation problems.

Reply:
I agree with you in most cases, but this is for testing so for me there is a lower threshold.

I think the ability to have this all in a single file outweighs putting the YAML in a second file. I can just copy this file anywhere I want and run it.  I don't have to keep track of two files.

Also, as soon as the YAML is split out, the next ask would be to split out the script into it's own file for the same reasons. Then it gets even more complicated.

I also have a second generator for Tekton pipelines coming. It's the same, all in a single file, and that's even more important for that one.  That requires some substitution to be done on the YAML files before they are applied. That's easily done with the YAML contained in the file.

Reply:
Another point is that with additional files we'll need to have each generator in its own sub-directory.  That's not a big deal really, but I think it cuts down on the clutter to have them all self-contained in one file.

Reply:
> Also, as soon as the YAML is split out, the next ask would be to split out the script into it's own file for the same reasons. Then it gets even more complicated.

You are right about this, I also thought of mentioning this and suggest using Kustomization for that, but, I prefered to suggest using an image that already generate logs for us.

> I also have a second generator for Tekton pipelines coming. It's the same, all in a single file, and that's even more important for that one. That requires some substitution to be done on the YAML files before they are applied. That's easily done with the YAML contained in the file.

I think that with this approach you are prioritizing "easy to run" (for the copy/paste reference that you say) than "easy to read". I prefer the second approach and I think that we usually prefer that for the code that we haven't wrote ourselves (think of the general ask of not putting to much logic on top of the helm chart to be able to have it more readable). 

I haven't seen the Tekton pipeline thing yet so I'm not able to provide an alternative to it.
About this PR, I still think that extracting the YAML shouldn't be a big deal, and we could defer the discussion about the Kustomize thing (we can even create a separate issue+pr for that and see how it looks, the refactor shouldn't take long and we can always close the PR if we see that doesn't make things more readable)